File: backend/src/modules/users/user.model.js
const mongoose = require('mongoose');

const userSchema = new mongoose.Schema({
  phone: {
    type: String,
    required: true,
    unique: true,
    index: true,
  },
  name: {
    type: String,
    trim: true,
  },
  email: {
    type: String,
    trim: true,
    lowercase: true,
  },
  role: {
    type: String,
    enum: ['student', 'owner', 'admin'],
    default: 'student',
  },
  collegeEmail: {
    type: String,
    trim: true,
    lowercase: true,
  },
  collegeName: {
    type: String,
    trim: true,
  },
  studentVerificationStatus: {
    type: String,
    enum: ['none', 'pending', 'verified', 'rejected'],
    default: 'none',
  },
  membershipTier: {
    type: String,
    enum: ['free', 'verified', 'premium'],
    default: 'free',
  },
  studentIdImage: {
    type: String,
  },
  savedAccommodations: [{
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Accommodation',
  }],
}, {
  timestamps: true,
});

module.exports = mongoose.model('User', userSchema);

File: mobile/lib/features/auth/domain/user_model.dart
class User {
  final String id;
  final String phone;
  final String? name;
  final String? email;
  final String role;
  final bool isPhoneVerified;
  final List<String> savedAccommodations;

  User({
    required this.id,
    required this.phone,
    this.name,
    this.email,
    required this.role,
    required this.isPhoneVerified,
    required this.savedAccommodations,
  });

  factory User.fromJson(Map<String, dynamic> json) {
    return User(
      id: json['_id'] ?? '',
      phone: json['phone'] ?? '',
      name: json['name'],
      email: json['email'],
      role: json['role'] ?? 'student',
      isPhoneVerified: json['isPhoneVerified'] ?? false,
      savedAccommodations: List<String>.from(json['savedAccommodations'] ?? []),
    );
  }
}
